File: Exercism Codes/JavaScript/atbash-cipher.js
export const encode = (text) => {
  const result = [];
  for (const x of text) {
    if (/\w/.test(x)) {
      result.push(transform(x));

      if ((result.length + 1) % 6 == 0) {
        result.push(" ");
      }
    }
  }
  return result.join("").trim();
};

export const decode = (text) => {
  return text.split("").filter(x => /\w/.test(x)).map(transform).join("");
};

const transform = (x) => {
  return /[A-Za-z]/.test(x) ? String.fromCharCode((26 - (x.toLowerCase().charCodeAt(0) - 97) - 1) + 97) : x;
};

File: Exercism Codes/C#/AtbashCipher.cs
public static class AtbashCipher
{
    public static string Encode(string plainValue)
    {
        List<char> result = [];
        foreach (char x in plainValue)
        {
            if (char.IsLetterOrDigit(x))
            {
                result.Add(Transform(x));

                if ((result.Count + 1) % 6 == 0)
                {
                    result.Add(' ');
                }
            }
        }
        return string.Join("", result).Trim();
    }

    public static string Decode(string encodedValue) => string.Join("", encodedValue.Where(x => char.IsLetterOrDigit(x)).Select(Transform));

    private static char Transform(char x) => char.IsLetter(x) ? (char)((26 - (char.ToLower(x) - 97) - 1) + 97) : x;
}
